File: app/src/main/java/com/aayush/filmipedia/view/activity/DetailActivity.java
package com.aayush.filmipedia.view.activity;

import android.content.Context;
import android.os.Bundle;

import com.aayush.filmipedia.R;
import com.aayush.filmipedia.model.Movie;
import com.aayush.filmipedia.view.fragment.DetailFragment;

import java.util.Objects;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;

import static com.aayush.filmipedia.util.Constants.EXTRAS_TAG;

public class DetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_detail);

        Movie movie = Objects.requireNonNull(getIntent().getExtras()).getParcelable(EXTRAS_TAG);
        Bundle bundle = new Bundle();
        bundle.putParcelable(EXTRAS_TAG, movie);

        Fragment fragment = DetailFragment.newInstance();
        fragment.setArguments(bundle);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.container, fragment)
                .commit();
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }
}

File: app/src/main/java/com/aayush/filmipedia/view/activity/DetailActivity.kt
package com.aayush.filmipedia.view.activity

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.aayush.filmipedia.R
import com.aayush.filmipedia.model.Movie
import com.aayush.filmipedia.util.TAG_EXTRAS
import com.aayush.filmipedia.view.fragment.DetailFragment
import io.github.inflationx.viewpump.ViewPumpContextWrapper

class DetailActivity: AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail)

        val movie = intent?.extras?.getParcelable<Movie>(TAG_EXTRAS)

        val fragment = DetailFragment.newInstance().apply {
            arguments = Bundle().apply { putParcelable(TAG_EXTRAS, movie) }
        }
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.container, fragment)
            .commit()
    }

    override fun attachBaseContext(newBase: Context) = super.attachBaseContext(ViewPumpContextWrapper.wrap(newBase))
}
